src/Grammar.java

```diff
@@ -0,0 +1,91 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class Grammar {
+    private List<String> nonTerminals;
+    private Set<String> terminals;
+    private List<Production> productions;
+    private String filename;
+
+    public Grammar(String filename) {
+        nonTerminals = new LinkedList<>();
+        terminals = new HashSet<>();
+        productions = new ArrayList<>();
+        this.filename = filename;
+        getGrammarFromFile();
+    }
+
+    private void getGrammarFromFile() {
```

Comment:
Don't have business login in simple DTO's please, this way you can create stateless services easier